src/main/kotlin/org/kamiblue/client/module/modules/movement/Anchor.kt

```diff
@@ -0,0 +1,92 @@
+package org.kamiblue.client.module.modules.movement
+
+import net.minecraft.util.math.BlockPos
+import net.minecraft.util.math.MathHelper
+import org.kamiblue.client.event.SafeClientEvent
+import org.kamiblue.client.event.events.PlayerTravelEvent
+import org.kamiblue.client.manager.managers.CombatManager
+import org.kamiblue.client.module.Category
+import org.kamiblue.client.module.Module
+import org.kamiblue.client.util.combat.SurroundUtils
+import org.kamiblue.client.util.combat.SurroundUtils.checkHole
+import org.kamiblue.client.util.math.VectorUtils.toBlockPos
+import org.kamiblue.client.util.threads.safeListener
+
+@CombatManager.CombatModule
+internal object Anchor : Module(
+    name = "Anchor",
+    description = "Automatically stops when you are above hole",
+    category = Category.MOVEMENT,
+    modulePriority = 10
+) {
+    private val vRange by setting("Range", 5, 1..10, 1)
+    private val mode by setting("Mode", AnchorMode.BOTH)
+    private val turnOffAfter by setting("Turn Off After", false)
+    private val pitchTrigger by setting("Pitch Trigger", false)
+    private val pitch by setting("Pitch", 80, -90..90, 1, { pitchTrigger })
+    private val strict by setting("Strict", true)
+    private var prevInHole = false
+
+    private enum class AnchorMode {
+        BOTH, BEDROCK
+    }
+
+    /**
+     * Checks whether the specified block position is a hole or not
+     */
+    private fun SafeClientEvent.isHole(pos: BlockPos): Boolean {
+        val type = checkHole(pos)
+        return ((mode == AnchorMode.BOTH && type != SurroundUtils.HoleType.NONE) ||
+            (mode == AnchorMode.BEDROCK && type == SurroundUtils.HoleType.BEDROCK))
```

Comment:
```suggestion
        return (mode == AnchorMode.BOTH && type != SurroundUtils.HoleType.NONE) ||
            (mode == AnchorMode.BEDROCK && type == SurroundUtils.HoleType.BEDROCK)
```
Remove the extra parenthesis

Reply:
it's just for readability but ok

Reply:
```java
        return (mode == AnchorMode.BOTH && type != SurroundUtils.HoleType.NONE) ||
            (mode == AnchorMode.BEDROCK && type == SurroundUtils.HoleType.BEDROCK)
```
i think this looks better

Reply:
The parentheses are still redundant because && has higher precedence than ||

Reply:
i understand.
for me it's more readable with brackets but ok if you really want to remove as much redundant code as possible